nowcasting_dataloader/datasets.py

```diff
@@ -179,6 +187,182 @@ def __getitem__(self, batch_idx: int) -> dict:
         return batch
 
 
+class SatFlowDataset(NetCDFDataset):
+    """SatFlow dataset for filtering and splitting up output from NetCDFDataset properly"""
+
+    def __init__(
+        self,
+        n_batches: int,
+        src_path: str,
+        tmp_path: str,
+        configuration: Configuration,
+        cloud: str = "gcp",
+        required_keys: Union[Tuple[str], List[str]] = None,
+        history_minutes: Optional[int] = None,
+        forecast_minutes: Optional[int] = None,
+        normalize: bool = False,
+        add_position_encoding: bool = False,
+        add_satellite_target: bool = False,
+        add_hrv_satellite_target: bool = False,
+    ):
+        """
+        Netcdf Dataset
+
+        Args:
+            n_batches: Number of batches available on disk.
+            src_path: The full path (including 'gs://') to the data on
+                Google Cloud storage.
+            tmp_path: The full path to the local temporary directory
+                (on a local filesystem).
+            cloud:
+            required_keys: Tuple or list of keys required in the example for
+                it to be considered usable
+            history_minutes: How many past minutes of data to use, if subsetting the batch
+            forecast_minutes: How many future minutes of data to use, if reducing the amount
+                of forecast time
+            configuration: configuration object
+            cloud: which cloud is used, can be "gcp", "aws" or "local".
+            normalize: normalize the batch data
+            add_position_encoding: Whether to add position encoding or not
+            add_satellite_target: Whether to add future satellite imagery to the target or not
+            add_hrv_satellite_target: Whether to add future HRV satellite imagery to the target
+        """
+        super().__init__(
+            n_batches,
+            src_path,
+            tmp_path,
+            configuration,
+            cloud,
+            required_keys,
+            history_minutes,
+            forecast_minutes,
+            normalize,
+            add_position_encoding,
+        )
+
+        self.add_satellite_target = add_satellite_target
+        self.add_hrv_satellite_target = add_hrv_satellite_target
+
+        # SatFlow specific changes, i.e. which timestep to split on
+        self.current_timestep_index = (self.history_minutes // 5) + 1
+        self.current_timestep_index_30 = (self.history_minutes // 30) + 1
+
+    def __getitem__(self, batch_idx: int) -> Tuple[dict, dict]:
+        """
+        Satflow extension for the dataloader
+
+        Args:
+            batch_idx: Batch ID to load
+
+        Returns:
+            Tuple of dicts of torch.Tensors holding the data
```

Comment:
Please document what the two dicts in the tuple hold the input and the target, respectively.